README.md

```diff
@@ -2,29 +2,74 @@
-   merge commits` box is unchecked and `Automatically delete head branches` box
-   is checked.
-4. Go to https://github.com/giantswarm/upgrade-schedule-operator/settings/access and add
-   `giantswarm/bots` with `Write` access and `giantswarm/employees` with
-   `Admin` access.
-5. Add this repository to https://github.com/giantswarm/github.
-6. Create quay.io docker repository if needed.
-7. Add the project to the CircleCI:
-   https://circleci.com/setup-project/gh/giantswarm/upgrade-schedule-operator
-8. Change the badge (with style=shield):
-   https://circleci.com/gh/giantswarm/upgrade-schedule-operator.svg?style=shield&circle-token=TOKEN_FOR_PRIVATE_REPO
-   If this is a private repository token with scope `status` will be needed.
-
-[generate]: https://github.com/giantswarm/upgrade-schedule-operator/generate
+This operator is intended to automate scheduling, triggering and notification for upgrades of giant swarm workload clusters.
+
+## prerequisites
+
+Running this operator alone will enable the automatic scheduling and triggering of upgrades on a management cluster in any provider.
+However, on AWS, we use the following components together for observability and avoiding errors:
+
+- [`event-exporter-app`](https://github.com/giantswarm/event-exporter-app) for exporting the upgrade events and notifying stakeholders in their slack channel.
+  This requires adding a token for the slack channels to the MC config if it is not there yet.
+- [`aws-admission-controller`](https://github.com/giantswarm/aws-admission-controller) to validate the format of the upgrade annotations.
+  This could be added to another validating webhook.
+- [`cluster-operator`](https://github.com/giantswarm/cluster-operator) that carries out the actual upgrading process and emits upgrade events on the cluster.
+  This could be added to another controller.
+
+## how to schedule the upgrade
+
+To schedule your upgrade, simply add the following annotations to the `Cluster` CR to specify the time and desired version.
+```
+annotations:
+  alpha.giantswarm.io/update-schedule-target-release: 15.2.1
+  alpha.giantswarm.io/update-schedule-target-time: "15 Sep 21 08:00 UTC"
+```
+Please note that the release version has to be an existing release higher than the current release version.
+The time has to be given in RFC822 format and UTC.
+Furthermore, only times that are at least 16 minutes in the future but not more than 6 months are accepted.
+
+## what happens next
+
+For your scheduled upgrades you should be able to see the remaining time in the logs of the `upgrade-schedule-operator`
+
+```
+2021-09-14T17:23:29.605Z	INFO	controllers.Cluster	The scheduled update time is not reached yet. Cluster will be upgraded in 14h37m0s at 2021-09-15 08:00:00 +0000 UTC.	{"cluster": "default/xyz01"}
+```
+Around 10-15 minutes before the scheduled upgrade, a slack message should appear in the specified slack channel.
```

Comment:
Where is the slack channel specified? Can this vary by cluster?

Reply:
I thought there was some doc in the `event-exporter-app` repo but looks like there isn't. I believe you need to create a token for a channel to the `Giant Swarm Cluster Upgrade` app in slack and then add it to the config for the MC: https://github.com/giantswarm/config
I think how it's set up at the moment, we can have one per MC but sometimes they are shared for the different MCs of the same customer.